src/librustc/ty/layout.rs

```diff
@@ -356,33 +372,37 @@ impl Integer {
             (I16, false) => tcx.types.u16,
             (I32, false) => tcx.types.u32,
             (I64, false) => tcx.types.u64,
+            (I128, false) => tcx.types.u128,
             (I1, true) => tcx.types.i8,
             (I8, true) => tcx.types.i8,
             (I16, true) => tcx.types.i16,
             (I32, true) => tcx.types.i32,
             (I64, true) => tcx.types.i64,
+            (I128, true) => tcx.types.i128,
         }
     }
 
     /// Find the smallest Integer type which can represent the signed value.
     pub fn fit_signed(x: i64) -> Integer {
```

Comment:
I've edited the file to use i128 accordingly, but unfortunately it panicked in stage2: `thread 'rustc' panicked at 'assertion failed: bits <= 64', src/librustc_trans/adt.rs:388`.

I'm not that really happy about adding things that slow down merging of this PR, therefore I'd prefer if it were done in a follow up PR after this one, this PR doesn't add any breakage I guess...